pkg/conversions/pod_conversion.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package conversions
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// PodConverter implements the interface to convert k8s pod object to a stripped down
+// version of pod to save on memory utilized
+type podConverter struct {
+	K8sResource     string
```

Comment:
i don't think this two variable is needed.

Reply:
I feel internally we do need to store resource and resourceType. So Instead I made them private and changed them name so that only constructor initializes with predefined values